### PR TITLE
PLANET-1888 open graph markup

### PR DESCRIPTION
**Pull request by angtheod**
Add Facebook Open Graph and Twitter meta info for better sharing of Posts/Pages
(In the 1st commit had added info also for Campaign pages but removed it as it is not in the requirements).

### CODE REVIEW

File: templates/html-header.twig [hunk 1]
```diff
@@ -5,6 +5,28 @@
 	<title>{% if wp_title %}{{ wp_title|e('esc_html')|raw }} - {{ site.name }}{% else %}{{ site.name }}{% endif %}</title>
 	<meta name="description" content="{{ site.description }}">
 
+	{% if post %}
+		{% set description   = post.post_excerpt %}
+		{% set attachment_id = post.background_image_id ?? fn('get_post_thumbnail_id', post) %}
```
Comment (sagarsdeshmukh) [inline templates/html-header.twig]: @angtheod Just a question - can we move `post.background_image_id ?? fn('get_post_thumbnail_id', post) %}`  this logic to `page.php`?

**Comment by sagarsdeshmukh** [comment]
Please confirm if it works on home page for fields `og:title` , `og:image` , `og:image:width`,  `og:image:height`, `twitter:title`  and `twitter:image`.

File: templates/html-header.twig (hunk 1 above)
Comment (angtheod) [reply templates/html-header.twig]: @sagarsdeshmukh Why should we do this?

Comment (sagarsdeshmukh) [reply templates/html-header.twig]: I think we can keep the conditional part in page.php

**Comment by angtheod** [comment]
@sagarsdeshmukh Which image should be chosen for home page? There is no instruction/requirement for this as far as I know.

**Comment by sagarsdeshmukh** [comment]
While checking on home page, I got above fields blank, but honestly saying I am also not sure which image should appear here :)

**Comment by angtheod** [comment]
@sagarsdeshmukh Updated

File: single.php [hunk 2]
```diff
@@ -38,6 +38,7 @@ function add_body_classes_for_post( $classes ) {
 $page_terms_data             = get_the_terms( $post, 'p4-page-type' );
 $context['author_override']  = $page_meta_data['p4_author_override'][0] ?? '';
 $context['background_image'] = $page_meta_data['p4_background_image_override'][0] ?? '';
+$context['post_image_id']    = $page_meta_data['p4_background_image_override_id'][0] ?? $page_meta_data['_thumbnail_id'][0];
```
Comment (kirdia) [inline single.php]: I'm not sure, was this meant to be p4_background_image_override instead of p4_background_image_override_id?

Comment (angtheod) [reply single.php]: nop, I wanted the id

**Pull request merged**